### DISCUSSION TITLE
Support Karpenter On-Demand/Spot Ratio Split ?

### DISCUSSION THREAD

**Discussion by xubofei1983** [argoproj/argo-workflows#12235] [Q&A] [answered]
https://karpenter.sh/docs/concepts/scheduling/#on-demandspot-ratio-split

I am testing with argo-workflows with Karpenter scheduler.

My understanding is that it requires the argo workflow pod to set topologySpreadConstraints, in order for it to work.  Does anyone know whether argo supports it?

Thanks.

**Comment by agilgur5** [comment, marked as answer]
You provide a Pod spec to all Workflow steps, so that can include `topologySpreadConstraints`.

For the Controller itself, you probably want on an on-demand instance so that tracking is not interrupted.
